packages/ethers/src/index.ts

```diff
@@ -128,13 +134,23 @@ export class TurnkeySigner extends AbstractSigner implements ethers.Signer {
         activityType: type,
       });
     } else {
-      const activity = await this.client.signTransaction({
-        signWith: this.signWith,
-        type: "TRANSACTION_TYPE_ETHEREUM",
-        unsignedTransaction: message,
-      });
+      const { activity, signedTransaction } = await this.client.signTransaction(
+        {
+          signWith: this.signWith,
+          type: "TRANSACTION_TYPE_ETHEREUM",
+          unsignedTransaction,
+        }
+      );
 
-      return assertNonNull(activity?.signedTransaction);
+      if (activity.status !== "ACTIVITY_STATUS_COMPLETED") {
+        throw new TurnkeyActivityError({
+          message: `Unexpected activity status: ${activity.status}`,
+          activityId: activity.id,
+          activityStatus: activity.status as TActivityStatus,
+        });
+      }
```

Comment:
I think the key is to throw a _different_ error when the activity is in need of consensus. Something like:
```ts
if (activity.status == "ACTIVITY_STATUS_NEEDS_CONSENSUS") {
    throw new TurnkeyActivityNeedsConsensus({
          message: `Activity requires consensus`,
          activityId: activity.id,
          activityStatus: activity.status as TActivityStatus,
        });
}
```

Then I agree with @andrewkmin, the retry or error behavior can be up to the caller.

Reply:
ended up adding some bespoke errors to viem. the thing is it needs `.walk()` so I ended up extending the `BaseError` class. let me know what you think — if it adds unnecessary complexity, happy to cut it or reduce it down!